hunting/linux/queries/execution_uncommon_process_execution_from_suspicious_directory.toml

```
@@ -0,0 +1,44 @@
+[hunt]
+author = "Elastic"
+description = """
+This hunt identifies uncommon process executions from suspicious directories on Linux systems. It looks for processes executed from directories that are often associated with malicious activity, such as /dev/shm, /var/www, /boot, /srv, /tmp, /var/tmp, /run, and /var/run. The goal is to detect potentially malicious processes while excluding common, benign ones.
+"""
+integration = ["endpoint"]
+uuid = "0ea47044-b161-4785-ba99-e11f46d6ac51"
+name = "Uncommon Process Execution from Suspicious Directory"
+language = ["ES|QL"]
+license = "Elastic License v2"
+notes = [
+    "Excluded /tmp, /var/tmp, /run, /var/run subdirectories to reduce noise.",
+    "Excluded /tmp, /var/tmp files starting or ending with digits to exclude real temporary files.",
+    "Included a process or parent process count of <= 3, and a host count of <= 3 to eliminate common processes across different hosts.",
+]
+mitre = ["T1036.004", "T1049", "T1059", "T1059.004"]
+
+query = [
+'''
+from logs-endpoint.events.process-*
+| where @timestamp > now() - 7 day
+| where host.os.type == "linux" and event.type == "start" and event.action == "exec" and (
+// Add paths to monitor from your environment here
+  (process.executable like "/dev/shm/*") or
+  (process.executable like "/var/www/*") or
+  (process.executable like "/boot/*") or
+  (process.executable like "/srv/*") or
+  ((process.executable rlike "/tmp/[^/]+" or process.executable rlike "/var/tmp/[^/]+")) or
+  (process.executable rlike "/run/[^/]+") or
+  (process.executable rlike "/var/run/[^/]+")
+) and not (
+  // Exclude noisy (parent) processes, users or directories from your environment here
+  (process.parent.executable in ("/usr/sbin/dpkg-preconfigure")) or
+  // Exclude /tmp and /var/tmp instances starting or ending with digits (usually benign files)
+  (process.executable rlike "/tmp/[0-9].*" or process.executable rlike "/tmp/.*[0-9]/?") or
+  (process.executable rlike "/var/tmp/[0-9].*" or process.executable rlike "/var/tmp/.*[0-9]/?")
+)
+| stats cc = count(), host_count = count(host.name) by process.executable, process.parent.executable, host.name, user.id
```

the more you reduce by a, b, c, d to the minimum the more you improve performance. also aggregating by unique fields like user.name, source.ip etc is good if it align with logic like top noisy users or top targeted destinations, in this rule it's about the process.executable. 

```suggestion
| stats cc = count(), host_count = count_distinct(host.name) by process.executable, process.parent.executable
```